src/parsing/av3C.js

```diff
@@ -0,0 +1,546 @@
+		},
+		CondByteSwap16LE: function(val) {
+			return OSisLittleEndian ? val : this.ByteSwap16(val);
+		},
+		GetUInt16BE: function(val) {
+			return this.CondByteSwap16BE(val);
+		},
+		GetUInt16LE: function(val) {
+			return this.CondByteSwap16LE(val);
+		},
+		
+		getUint32: function() {
+			return this._big_endian ? this.GetUInt32BE(this.rdb(4)) : this.GetUInt32LE(this.rdb(4));
+		},
+		ByteSwap32: function(x) {
+			return (x << 24) | ((x << 8) & 0x00FF0000) | ((x >> 8) & 0x0000FF00) | (x >> 24);
+		},
+		CondByteSwap32BE: function(val) {
+			return OSisLittleEndian ? this.ByteSwap32(val) : val;
+		},
+		CondByteSwap32LE: function(val) {
+			return OSisLittleEndian ? val : this.ByteSwap32(val);
+		},
+		GetUInt32BE: function(val) {
+			return this.CondByteSwap32BE(val);
+		},
+		GetUInt32LE: function(val) {
+			return this.CondByteSwap32LE(val);
+		},
+	
+		getBits: function(bits) {
+			// No read if read error is already set or not enough bits to read.
+			if (this._read_error || this.currentReadBitOffset() + bits > this.currentWriteBitOffset()) {
+				this._read_error = true;
+				return 0;
+			}
+			var val = 0;
+			if (this._big_endian) {
+				// Read leading bits up to byte boundary
+				while (bits > 0 && this._state.rbit != 0) {
+					val = (val << 1) | this.getBit();
+					--bits;
+				}
+	
+				// Read complete bytes
+				while (bits > 7) {
+					val = (val << 8) | this._buffer[this._state.rbyte++];
+					bits -= 8;
+				}
+	
+				// Read trailing bits
+				while (bits > 0) {
+					val = (val << 1) | this.getBit();
+					--bits;
+				}
+			}
+			else {
+				// Little endian decoding
+				var shift = 0;
+	
+				// Read leading bits up to byte boundary
+				while (bits > 0 && this._state.rbit != 0) {
+					val |= (this.getBit() << shift);
+					--bits;
+					shift++;
+				}
+	
+				// Read complete bytes
+				while (bits > 7) {
+					val |= this._buffer[this._state.rbyte++] << shift;
+					bits -= 8;
+					shift += 8;
+				}
+	
+				// Read trailing bits
+				while (bits > 0) {
+					val |= (this.getBit() << shift);
+					--bits;
+					shift++;
+				}
+			}
+			return (val);
+		},
+		skipBits: function(bits) {
+			if (this._read_error) {
+				// Can't skip bits and bytes if read error is already set.
+				return false;
+			}
+			var rpos = 8 * this._state.rbyte + this._state.rbit + bits;
+			var wpos = 8 * this._state.wbyte + this._state.wbit;
+			if (rpos > wpos) {
+				this._state.rbyte = this._state.wbyte;
+				this._state.rbit = this._state.wbit;
+				this._read_error = true;
+				return false;
+			}
+			else {
+				this._state.rbyte = rpos >> 3;
+				this._state.rbit = rpos & 7;
+				return true;
+			}
+		},
+	
+		getUE: function() {
+			// read in an unsigned Exp-Golomb code;
+	
+			if (this.getBit() == 1)
+				return 0;
+			else {
+				var zero_count=1;
+				while (this.peekBit() == 0) {
+					this.getBit();
+					zero_count++;
+				}
+				var tmp_value = this.getBits(zero_count+1);
+				return tmp_value - 1;
+			}
+		}
+	};
```

Comment:
Why not reuse the existing code that reads bits/bytes?

Reply:
From what I could see, the existing code only read byte aligned values - the AVS3 sequence header is just a bitstream.
This bit reader is based on the bitstream reader from TSduck.